packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -112,119 +156,98 @@ export class SemanticViewerResultsRunner implements IResultsRunner {
         );
     }
 
-    getDimensions(): VizIndexLayoutOptions[] {
-        return this.dimensions.map((dimension) => ({
-            reference: dimension.name,
-            axisType: getVizIndexTypeFromDimensionType(dimension.type),
-            dimensionType: getDimensionTypeFromSemanticLayerFieldType(
-                dimension.type,
-            ),
-        }));
-    }
-
-    getMetrics(): VizValuesLayoutOptions[] {
-        return this.metrics.map((metric) => ({
-            reference: metric.name,
-            aggregation: metric.aggType || VIZ_DEFAULT_AGGREGATION,
-        }));
+    async getPivotedVisualizationData(
+        query: SemanticLayerQuery,
+    ): Promise<PivotChartData> {
+        if (!!query.pivot?.index.length || !!query.pivot?.values.length) {
+            return {
+                results: [],
+                indexColumn: undefined,
+                valuesColumns: [],
+                columns: [],
+            };
+        }
+        return this.runPivotQuery(query);
     }
 
     getPivotQueryDimensions(): VizIndexLayoutOptions[] {
+        // the same as pivotChartIndexLayoutOptions
         return this.dimensions.map((dimension) => ({
             reference: dimension.name,
-            axisType: getVizIndexTypeFromDimensionType(dimension.type),
+            axisType: getVizIndexTypeFromSemanticLayerFieldType(dimension.type),
             dimensionType: getDimensionTypeFromSemanticLayerFieldType(
                 dimension.type,
             ),
         }));
     }
 
     getPivotQueryMetrics(): VizValuesLayoutOptions[] {
```

Comment:
we don't need this to vary per runner - it supports both. sql runner simply has no metrics

Reply:
Fine by me. My only counter thought is that having it in both places makes it very clear what you'd override in the future if the implementations diverge. But we can get to that then. 

This is fine for now since we are combining the runners.